### DISCUSSION TITLE
Early return (control flow)

### DISCUSSION THREAD

**Discussion by E8y2FqZE** [sqlpage/SQLPage#371] [Q&A] [answered]
Is there a good way to do control flow to stop rendering further components if a condition is encountered? For example, if a user is unauthorized, I'd rather display a message and stop rendering further components.

```
set user_id = (select user_id from sessions where id = sqlpage.cookie('session'));

-- render a message if appropriate
select 'title' as component,
'Not authenticated' as contents
where $user_id is null;

-- exit here if unauthenticated
if $user_id is null RETURN; -- (fake SQL)


-- when successfully authenticated, show this stuff without having to add "where $user_id is not null" to the end of each query
select 'list' as component, ...
select 'card' as component, ...
select 'carousel' as component ...
select 'form' as component, ...
select 'form' as component, ...
select 'title' as component ...
```

**Comment by lovasoa** [comment, marked as answer]
Hello! Yes, you can use the redirect component for this. It will stop executing the current page, and redirect to another one (an error page in your case).

**Reply by lovasoa** [reply]
You can have a look at how this is done using the authentication example in this repo.
